### PR TITLE
[WIP]: Cleaner OIDC integration (#13592), Custos enhancements

### PR DESCRIPTION
**Pull request by ahmedhamidawan**
## Enhancement 1: Cleaner OIDC integration
Adds the enhancement from https://github.com/galaxyproject/galaxy/issues/13592, i.e.: a cleaner OIDC integration.
I have added a config boolean: `oidc_auto_redirect`, which when set true (along with `enable_oidc`) will only show the OIDC login option (and no Galaxy login or registration option) on the login page. I hope this resolves that issue.

## Enhancement 2: Custos login ambiguity enhancement
Custos is an OIDC provider that uses CLogon to allow user to login/register using their institutional accounts. However, there is an issue where a user might have an existing internal Galaxy account with the same email as their institutional account (e.g.: They made an account with a jh.edu email, then used Custos to login with the same JHU account). This can create duplicates in the database and is just a cause for ambiguity.
So, in this enhancement, on the registration page, when `enable_oidc` is `true`, I have added accordion `b-collapse`s that encourage the user to, instead of creating a local Galaxy account, use their Custos login to create an account. If they do not prefer that, there is still the `b-collapse` for local registration.
This has been demonstrated here:

https://user-images.githubusercontent.com/78516064/161148051-8b71a24e-a8c5-423d-a6c9-f280f2c1596b.mov

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/config/schemas/config_schema.yml [hunk 1]
```diff
@@ -2630,6 +2630,13 @@ mapping:
         desc: |
           Enables and disables OpenID Connect (OIDC) support.
 
+      oidc_auto_redirect:
+        type: bool
+        default: false
+        required: false
+        desc: |
+          Enables and disables auto-redirect to OpenID Connect (OIDC) login.
```
Comment (nsoranzo) [inline lib/galaxy/config/schemas/config_schema.yml:2638]:
```suggestion
          Enable auto-redirect to OpenID Connect (OIDC) login.
```

**Comment by dannon** [comment]
@ahmedhamidawan So the derivative property here (to automatically redirect when certain conditions are met) is still working fine for me, and I'm hesitant to add an overlapping configuration option as a first pass.  If I remember correctly, `oidc_auto_redirect` should more or less just be 1) do you have oidc enabled, and 2) is that the only option?  If so, we redirect.  

Did you manage to figure out from the user what exactly is going wrong with the existing approach?  My concern here is that an extra property just complicates configuration and leaves room for other corner cases in config mismatch due to things not being exactly defined, I think.

**Comment by ahmedhamidawan** [comment]
Only closed this because it's a WIP and for some reason it keeps triggering tests every now and then.
